p2p/protocol/identify/obsaddr_test.go

```diff
@@ -117,3 +118,48 @@ func TestObsAddrSet(t *testing.T) {
 		t.Error("addrs should have timed out")
 	}
 }
+
+func TestAddAddrsProfile(b *testing.T) {
+	m := func(s string) ma.Multiaddr {
+		m, err := ma.NewMultiaddr(s)
+		if err != nil {
+			b.Fatal(err)
+		}
+		return m
+	}
+	oas := &ObservedAddrSet{}
+
+	add := func(oas *ObservedAddrSet, observed, observer ma.Multiaddr) {
+		dummyLocal := m("/ip4/127.0.0.1/tcp/10086")
+		dummyDirection := net.DirOutbound
+
+		oas.Add(observed, dummyLocal, observer, dummyDirection)
+	}
+
+	a1 := m("/ip4/1.2.3.4/tcp/1231")
+	a2 := m("/ip4/1.2.3.4/tcp/1232")
+	a3 := m("/ip4/1.2.3.4/tcp/1233")
+	a4 := m("/ip4/1.2.3.4/tcp/1234")
+	a5 := m("/ip4/1.2.3.4/tcp/1235")
+
+	b1 := m("/ip4/1.2.3.6/tcp/1236")
+	b2 := m("/ip4/1.2.3.7/tcp/1237")
+	b3 := m("/ip4/1.2.3.8/tcp/1237")
+	b4 := m("/ip4/1.2.3.9/tcp/1237")
+	b5 := m("/ip4/1.2.3.10/tcp/1237")
+
+	_ = []ma.Multiaddr{a1, a2, a3, a4, a5, b1, b2, b3, b4, b5}
+
+	var wg sync.WaitGroup
+	for i := 0; i < 1000; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for j := 0; j < 10000; j++ {
```

Comment:
```suggestion
			for j := 0; j < 1000; j++ {
```

(so this can actually pass on travis)